File: applications/summersim18/wrapped/cpp/mul-showcase/right-mul.cpp
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <float.h>
#include "../../../../../cpp/numb/numb.hpp"

int main()
{
    Numb::setup("mul-cache2", "mul-cache1", 20, "use-cache");
    Numb a(0.001);
    Numb b(1e308);
    Numb c(11);

    Numb d = a * (b * c);
    cout << "0.001 * (1e308 * 11) = " << d << endl;

    return 0;
}

File: applications/summersim18/wrapped/python/mul-showcase/right-mul.py
from numb import *

if __name__ == '__main__':
    Numb.setup(cache_out='mul-cache2', cache_in='mul-cache1', precision=20, strategy='use-cache')
    a = Numb(0.001)
    b = Numb(1e308)
    c = Numb(11)
    
    d = a * (b * c)
    print "0.001 * (1e308 * 11) = {0}".format(d)
